kyo-prelude/shared/src/test/scala/kyo/LayerTest.scala

```diff
@@ -40,14 +40,51 @@ class LayerTest extends Test:
             val sharkLayer     = Layer.from(g => Shark(g))
             val megaSharkLayer = Layer.from(s => MegaShark(s))
 
-            val combinedLayer = guppyLayer using sharkLayer using megaSharkLayer
+            val combinedLayer = guppyLayer provide sharkLayer provide megaSharkLayer
 
             val env = Memo.run(combinedLayer.run).eval
             assert(env.get[Guppy].name == "Tiny Guppy")
             assert(env.get[Shark].belly.name == "Tiny Guppy")
             assert(env.get[MegaShark].belly.belly.name == "Tiny Guppy")
         }
 
+        "a using b with Env2" in {
+            trait Dummy
+            case class Guppy(name: String)
+            case class Shark(belly: Guppy, dummy: Dummy)
+
+            val guppyLayer: Layer[Guppy, Any] = Layer(Guppy("Tiny Guppy"))
+            val sharkLayer                    = Layer.from(Shark.apply)
+            val dummy: Layer[Dummy, Any]      = Layer(new Dummy {})
+
+            val guppyAndShart: Layer[Guppy & Shark, Env[Dummy]] = guppyLayer provide sharkLayer
+            // note: circular deps, not usable
+            val _: Layer[Shark & Guppy, Env[Guppy & Dummy]] = sharkLayer using guppyLayer
```

Comment:
TODO: write a pendingUntilFixed, that composition should have been detected, and should not compile